File: Week_02/299.getHint.js
var getHint = function(secret, guess) {
    let countA=0
    let countB=0
    let hashmap=new Map()
    let arr=[]

    for(let i=0;i<secret.length;i++){
        if(secret[i]==guess[i]){
            countA++
        }else{
            hashmap.has(secret[i])?hashmap.set(secret[i],hashmap.get(secret[i])+1):hashmap.set(secret[i],1)
            arr.push(guess[i])
        }
    }
    for(let i=0;i<arr.length;i++){
        if(hashmap.has(arr[i])){
            hashmap.get(arr[i])>1?hashmap.set(arr[i],hashmap.get(arr[i])-1):hashmap.delete(arr[i])
            countB++
        }
    }
    return countA+"A"+countB+"B"
};

File: Week_02/299-getHint.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        countA=0
        countB=0
        dic1={}
        arr=[]
        siz=len(secret)
        for i in range(siz):
            if secret[i]==guess[i]:
                    countA+=1
            else:
                if secret[i] in dic1:
                    dic1[secret[i]]+=1
                else:
                    dic1[secret[i]]=1
                    arr.append(guess[i])
        for i in range(len(arr)):
            if arr[i] in dic1:
                if dic1[arr[i]]>1:
                    dic1[arr[i]]-=1
                else:
                    dic1.pop(arr[i])
                countB+=1
        return str(countA)+"A"+str(countB)+"B"
